src/converters/replugged.ts

```diff
@@ -1,13 +1,52 @@
+import { ClassDeclaration, ClassMethod, ExportDefaultDeclaration, Identifier, Statement } from "@babel/types";
 import { IModImplementation } from "../api/ModImplementation.js";
 import { createFunctionFromObjectProperty } from "../api/RuntimeGenerators.js";
 import { IBaseWebpackApi } from "../api/Webpack.js";
+import { parse } from "@babel/parser";
 
 class RPWebpackApi implements IBaseWebpackApi {
     get getModule() {
         return createFunctionFromObjectProperty("replugged.webpack", "getModule");
     }
 }
 
+export function convertFormat(ast: Statement[]) {
+    // debugger;
+    // need to add this:
+    /*
+    const instance = new CrossCompiledSample();
+    export const start = instance.start.bind(instance);
+    export const stop = instance.stop.bind(instance);
+    */
+    let targetClassName = undefined;
+    for (let index2 = 0; index2 < ast.length; index2++) {
```

Comment:
I personally would do something like this

```ts
let targetClassName;

for (const node of ast) {
    if (node.type !== "ExportDefaultDeclaration") continue;

    const exportDeclaration = node as ExportDefaultDeclaration;
    if (exportDeclaration.declaration.type !== "ClassDeclaration") continue;

    const exportedClass = exportDeclaration.declaration as ClassDeclaration;
    const hasStartMethod = exportedClass.body.body.some(
        (method) => (method as ClassMethod).key?.name === "start"
    );

    if (!hasStartMethod) continue;

    targetClassName = exportedClass.id?.name;
    break;
}
```

Reply:
Re-reading this i noticed that if the exported class has a `start` method but not an id it breaks but why is id optional?

Reply:
what do you mean?

Reply:
> what do you mean?

`targetClassName = exportedClass.id?.name;`

Why the `?`

Reply:
according to our format the entrypoint class must have a name. `exportedClass.id?.name;` essentially means if `exportedClass` doesn't have `id` (doesn't have an Identifier, in other words, has no name), `targetClassName` shall be `undefined`. otherwise (when `exportedClass` has a name), `targetClassName` should be set to the class name

Reply:
Oh my bad :+1: